src/orderbook/state.ts

```diff
@@ -146,6 +299,44 @@ export class AccountState {
     return account
   }
 
+  /**
+   * Updates a user's account balance.
+   */
+  private updateBalance(
+    user: Address,
+    token: TokenId | Address,
+    update: (amount: bigint) => bigint,
+  ): void {
+    const tokenAddr = this.tokenAddr(token)
+    const balances = this.account(user).balances
+    const newBalance = update(balances.get(tokenAddr) || BigInt(0))
+    balances.set(tokenAddr, newBalance)
+  }
+
+  /**
+   * Sets a pending withdrawal for a user for the specified token and amount.
+   */
+  private setPendingWithdrawal(
+    user: Address,
+    token: TokenId | Address,
+    batchId: number,
+    amount: bigint,
+  ): void {
+    const tokenAddr = this.tokenAddr(token)
```

Comment:
nit: should we assert that it's not already set or if so in the same batchID?

Reply:
I was under the impression you could set pending withdrawals multiple times and that it would just clear the previous one.

Reply:
If both requests happen in the same batch, yes. In a later batch it will first execute the matured withdraw and then request the new one.

Reply:
I see! I got scared for a moment but I realized that it also emits a `Withdraw` event.